src/steth/Swapper.sol

```diff
@@ -126,25 +131,41 @@ contract Swapper {
         return amountReceived;
     }
 
-    function lidoSwapWethToWstEth(uint256 _wethAmount) external {
+    /**
+     * Swap WETH to wstETH using Lido or Curve for ETH to stETH conversion, whichever is cheaper.
+     * @param _wethAmount Amount of WETH to swap.
+     * @return Amount of wstETH received.
+     */
+    function lidoSwapWethToWstEth(uint256 _wethAmount) external returns (uint256) {
         // weth to eth
         weth.withdraw(_wethAmount);
 
-        // stake to lido / eth => stETH
-        stEth.submit{value: _wethAmount}(address(0x00));
+        // eth to stEth
+        // if curve exchange rate is better than lido's 1:1, use curve
+        if (curvePool.get_dy(0, 1, _wethAmount) > _wethAmount) {
+            curvePool.exchange{value: _wethAmount}(0, 1, _wethAmount, 0);
```

Comment:
exchanging with a minimum out of 0 here?

Reply:
the function call above will give us the exact amount for the swap, but I guess it won't hurt to have this changed